### ISSUE TITLE
Okay. full description on bloxeed RNG gameflow.

### ISSUE THREAD

**Issue by zaphod77** [2Tie/TGMsim#46]
At mode select, there's 1 pull to seed piece sequence.  second pull is for player 1's first piece's item (which cant' ever happen)  third pull is for second player's first piece's item even if the second player isn't playing yet) the fourth pull would then be for the first player to start's second piece, if playe 2 starts the item seed will advance once, when player2's second piece is generated.

Given that, to actually generate the poweron item sequence, the starting seed for that should be 6037B943, assuming game windows do not interact in TGMSIm.

Timing the 8th item for a powerup block at height below 9 (not too difficult) gives you a Flicky this way.

Yes, this does mean that the powerups are actually determined by your piece sequence the first time through without continuing, and that your next game if nto cotinued gets a piece seequence that depends on how many pieces yo place.  an incredibly clever player could do some silly stuff by playing a specific number of pieces for each game before their maxout attempt.

**Comment by 2Tie** [comment]
related to #45 , will be solved at the same time.

**Comment by zaphod77** [comment]
when i grabbed it and compiled. i only got gem blocks, and they would do nothing and vanish after locking down.

So it seems the power up blocks themselves need implementing as well.

There is quite a bit of subtlety to many of them.  

1) you can fire every frame by alternating two rotation buttons every frame for satellite and flicky.  the arcade WILL track every press, provided you don't press more than once a frame.
2) four is the easiest to get right, as it never changes.
3) bomb and 16 ton weight share code, to determine their rate of falling. it may be the same as flicky and satellite, but i'm not sure.
4) flicky and satellite have proper lock delay, and it appears exactly the same as a piece falling with their gravity (which is not the same as current piece gravity). bomb and 16 ton weight essentially go off when they would overlap a cell in a pre lock delay tetris game.

Also, if you restart instead of return to menu, it should reuse the piece sequence in Bloxeed, but generate new items. :)

Flicky in particular can be used to score an actual tetris. the first one that lears a lie will be a sigle, but i you drop the five more before the single finishes clearing you will get a tetris.  Due to a bug, 5 line clears are scored as 1, 6 as 2, etc.  the table wraps around though the line counter goes up the right amount.  Considering how high scoring would have to get with the progression, perhaps it's best this bug is i place. flicky is OP enough even with the bug.

**Comment by 2Tie** [comment]
correct, items themselves are not implemented at this time (and have been on my todo list for six years now), but the tables and randomizer should be correct for generating them.

**Comment by zaphod77** [comment]
Flicky is probably the hardest one to implement, because line clears are checked and happen while it's on the field.  Once you have that one,  the other five are easy. let me try and get some data to help.

For the 4 item, it's 10 frames between the lines getting filled in for it. they then get all cleared at the same time, with the usual line clear delay.

Item block vanishes on first frame of line clear. Explosion is 30 frames. item score starts floating up one frame later. on frame 13 the item appears above the playfield (3 after the lines drop down). it stays above the field 182 frames (about) then enters the field on the 183 frame.  Pressing left or right doesn't cause the powerups to enter the frame, but pressing down DOES, and so does pressing A B or C. the press of the buttons that caused flicky or satellite to enter does NOT fire. 

gravity table for falling powerups. seems unaffected by difficulty level.

0=51
1=51
2=51
3=41
4=41 (not yet confirmed, but all the others are, and it's the only value that makes sense here)
5=31
6-21 (max until after 10)
7=21
8=21
9=21
10=31 (break at 10)
11=21 (speeds up every level now)
12=19 (-2 frames per level)
13=17
14=15
15+=13 (until you get to  level 15, which maxes it out at 13 frames)

more preliminary testing suggests that while flicky can drop blocks at 60 Hz with alternating buttons, satellite cannot actually fire at 60 hz.

the flicky drop code is strange. the blocks visibly falls at a rate of one cell  pre frame but firing resets this animation.  You can buffer up to 8 with 60 hz firing, then it stops responding until all 8 have dropped, which takes the correct amount of time for the 8th block to drop, then it accepts inputs for the next batch.

**Comment by zaphod77** [comment]
Satellite can similarly fire 8 shots at once.   

Flicky dips down one row when firing blocks, and when the block returns to his hand (all blocks have stopped falling), Flicky goes back up one row.

**Comment by zaphod77** [comment]
once the bottom of 16 ton weight touches the stack, it cannot be moved, though it will pause for the lock delay time.

If the bomb touches the stack, it can only be moved if half of the bomb is touching the stack, and only if it could move down afterwards. If both halves of the bottom rest on the stack, it sticks like the weight.
